cilium-health/probeserver/Makefile

```diff
@@ -0,0 +1,18 @@
+include ../../Makefile.defs
+
+TARGET=cilium-health-probe
```

Comment:
nit: If I see a process in my process list called `cilium-health-probe`, I might think that this process is actively probing other things. But it's really just a simple HTTP server respoding to `/health`. Do you think something like `cilium-health-responder` would be more clear? (Other parts of the PR would need updating for such a change, too).